python/test/unit/language/test_core.py

```diff
@@ -6422,6 +6422,70 @@ def test_convert2d(M, N, src_layout, interm_layout, dst_layout, dtype, device, t
     torch.testing.assert_close(z, x, rtol=0, atol=0)
 
 
+src_layouts = [BlockedLayout([1, 1], [1, 64], [1, 1], [0, 1], [1, 1], [1, 1], [0, 1])]
+
+dst_layouts = [
+    LinearLayout([[0, 32]], [[0, 1], [0, 2], [0, 4], [0, 8], [0, 16], [1, 0]], [], []),
+    LinearLayout([[1, 0], [0, 32]], [[0, 1], [0, 2], [0, 4], [0, 8], [0, 16], [2, 0]], [], []),
+    LinearLayout([[1, 0], [2, 0], [0, 32]], [[0, 1], [0, 2], [0, 4], [0, 8], [0, 16], [4, 0]], [], [])
+]
+
+
+@pytest.mark.parametrize("M, dst_layout", [[2, dst_layouts[0]], [4, dst_layouts[1]], [8, dst_layouts[2]]])
+@pytest.mark.parametrize("src_layout", [src_layouts[0]])
+@pytest.mark.parametrize("N", [64])
+@pytest.mark.parametrize("dtype", ['float8e5', 'float16', 'float32', 'int64'])
+def test_convert_permlane_swap(M, N, src_layout, dst_layout, dtype, device, tmp_path: pathlib.Path):
+    if not is_hip_cdna4():
+        pytest.skip()
```

Comment:
Would be nice to provide a good skip message like "CDNA4 permlane swap specific tests"